components/Header.js

```diff
@@ -1,59 +1,122 @@
+import backIcon from "../public/images/back_button.svg";
+import lupe from "../public/images/lupe.svg";
 
 export default function Header({
   onMapShown,
   onLocateMe,
   usedLocateMe,
   usedMapShown,
+  usedSearchAround,
+  onSearchAround,
 }) {
+  const [showInput, setShowInput] = useState(false);
+
   function onLocate() {
-    console.log("usedLocateMe");
     onLocateMe();
   }
 
+  function onInput() {
+    setShowInput(true);
+  }
+
+  function handleSearchAroundData(event) {
+    event.preventDefault();
+    let searchInput = event.target.elements.ort.value;
+    setShowInput(false);
+    onSearchAround(searchInput);
+  }
+
+  function back() {
+    setShowInput(false);
+  }
+
   return (
     <>
-      <StyledHeader>
-        {usedLocateMe ? (
-          <Image
-            onClick={onLocate}
-            src={UsedLocateMe}
-            alt="UmMichHerum"
-            width={91}
-            height={51}
-          ></Image>
-        ) : (
-          <Image
-            onClick={onLocate}
-            src={UmMichHerum}
-            alt="UmMichHerum"
-            width={91}
-            height={51}
-          ></Image>
-        )}
-        {usedMapShown ? (
-          <Image
-            onClick={onMapShown}
-            src={menue}
-            alt="Menü_Btn"
-            width={39}
-            height={53}
-          ></Image>
-        ) : (
-          <Image
-            onClick={onMapShown}
-            src={usedMenue}
-            alt="Menü_Btn"
-            width={39}
-            height={53}
-          ></Image>
-        )}
-      </StyledHeader>
+      {showInput ? (
+        <StyledHeader>
+          <StyledForm onSubmit={handleSearchAroundData}>
+            <StyledButton type="button" onClick={back}>
+              <Image
+                src={backIcon}
+                alt="addMessageButton"
+                width={30}
+                height={30}
+              />
+            </StyledButton>
+            <StyledInputText
+              type="text"
+              name="ort"
+              placeholder="Gib einen Ort ein"
+            ></StyledInputText>
+            <StyledButton type="submit">
+              <Image src={lupe} alt="addMessageButton" width={30} height={30} />
+            </StyledButton>
+          </StyledForm>
+        </StyledHeader>
+      ) : (
+        <StyledHeader>
+          {usedLocateMe ? (
+            <Image
+              onClick={onLocate}
+              src={UsedLocateMe}
+              alt="UmMichHerum"
+              width={91}
+              height={51}
+            ></Image>
+          ) : (
+            <Image
+              onClick={onLocate}
+              src={UmMichHerum}
+              alt="UmMichHerum"
+              width={91}
+              height={51}
+            ></Image>
+          )}
+          {usedSearchAround ? (
+            <Image
+              onClick={onInput}
+              src={usedSearchAroundIcon}
+              alt="Menü_Btn"
```

Comment:
I think the alt-tag has to be readable so that a screen reader can read it out loud. I would change this to "Menü-Button" or something like that. Also at some alts you use english words and at some german. So maybe have a look at that some time :)